crates/exex/src/hints.rs

```diff
@@ -112,3 +119,188 @@ pub fn add_segment_hint() -> Hint {
+            let mut buf = Vec::new();
+            transaction.encode_without_signature(&mut buf);
+
+            block_body.extend([
+                // The length of the RLP encoded transaction
+                MaybeRelocatable::from(Felt252::from(buf.len())),
+                // The RLP encoded transaction
+                MaybeRelocatable::from(Felt252::from_bytes_be_slice(&buf)),
+                // The length of the signature payload
+                MaybeRelocatable::from(Felt252::from(transaction.signature.payload_len())),
+                // The signature payload
+                MaybeRelocatable::from(Felt252::from_bytes_be_slice(
+                    &transaction.signature.to_bytes(),
+                )),
+            ]);
+        });
+
+        block_body
+    }
+}
+
+pub fn block_hint(block: SealedBlock) -> Hint {
+    Hint::new(
+        String::from("block"),
+        move |vm: &mut VirtualMachine,
+              _exec_scopes: &mut ExecutionScopes,
+              ids_data: &HashMap<String, HintReference>,
+              ap_tracking: &ApTracking,
+              _constants: &HashMap<String, Felt252>|
+              -> HintExecutionResult {
+            // We retrieve the `model.Block*` pointer from the `ids_data` hashmap.
+            // This pointer is used to store the block-related values in the VM.
+            let block_ptr = get_ptr_from_var_name("block", vm, ids_data, ap_tracking)?;
+
+            vm.load_data(block_ptr, &block.to_cairo_vm_block()).map_err(HintError::Memory)?;
+
+            Ok(())
+        },
+    )
+}
+
+#[cfg(test)]
+mod test {
+    use std::path::PathBuf;
+
+    use super::*;
+    use cairo_vm::{
+        cairo_run::{cairo_run, CairoRunConfig},
+        types::layout_name::LayoutName,
+    };
+    use reth_testing_utils::generators::{self, random_block, BlockParams, Rng};
+
+    #[test]
+    fn test_block_hint() {
+        for _ in 0..10 {
+            // Generate a random block
+            let mut rng = generators::rng();
+            let parent = rng.gen();
+            let tx_count = Some(rng.gen::<u8>());
+            let withdrawals_count = Some(rng.gen::<u8>());
+            let ommers_count = Some(rng.gen::<u8>());
+            let requests_count = Some(rng.gen::<u8>());
+            let block = random_block(
+                &mut rng,
+                10,
+                BlockParams {
+                    parent: Some(parent),
+                    tx_count,
+                    withdrawals_count,
+                    ommers_count,
+                    requests_count,
+                },
+            );
+
+            // Create a new cairo run configuration with the all_cairo layout
+            let config = CairoRunConfig { layout: LayoutName::all_cairo, ..Default::default() };
+
+            // Create a new hint processor with the block hint
+            let mut hint_processor =
+                KakarotHintProcessor::default().with_block(block.clone()).build();
```

Comment:
noob question but why do we need to `.clone()` since we don't reuse the `block` afterwards?

Reply:
Yes useless here, can be deleted, done.